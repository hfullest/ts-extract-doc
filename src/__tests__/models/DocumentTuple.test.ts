import { resolve } from 'path';
import { genrateDocument } from '../..';

describe('DocumentTuple', () => {
  const sourcePath = resolve(__dirname, '../fixtures/tuple.ts');

  it('文档模型解析[toTypeString]:', () => {
    const documents = genrateDocument(sourcePath);
    const result = documents.map((doc) => doc.map((it) => it.toTypeString())?.join('\n\n')).join('\n\n\n');
    expect(result).toMatchSnapshot();
  });

  it('快照', () => {
    const documents = genrateDocument(sourcePath);
    const result = documents.map((doc) => doc.map((it) => it.toTypeString())?.join('\n\n')).join('\n\n\n');
    expect(result).toMatchSnapshot();
  });
});
