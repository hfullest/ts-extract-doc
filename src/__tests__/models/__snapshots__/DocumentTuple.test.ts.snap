// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentTuple 快照 1`] = `"[{ a: string; readonly b?: number; },string,UiButtonProps<false>,(p1: number, p2?: string) => void,Function]"`;

exports[`DocumentTuple 文档模型解析[toTypeString]: 1`] = `"[{ a: string; readonly b?: number; },string,UiButtonProps<false>,(p1: number, p2?: string) => void,Function]"`;
